src/scanner.ts

```diff
@@ -39,69 +37,118 @@ export enum ScannerImageFormat {
 export class FujitsuScanner implements Scanner {
   public constructor(private format = ScannerImageFormat.PNG) {}
 
-  public async scanInto({
-    directory,
-    prefix,
-    onFileScanned,
-  }: ScanIntoOptions): Promise<void> {
+  public scanSheets(directory = dirSync().name): AsyncGenerator<Sheet> {
     const args = [
       '-d',
       'fujitsu',
       '--resolution',
       '300',
       `--format=${this.format}`,
       '--source=ADF Duplex',
-      '--swskip',
-      '0.5',
       '--dropoutcolor',
       'Red',
-      `--batch=${join(
-        directory,
-        `${prefix}${dateStamp()}-ballot-%04d.${this.format}`
-      )}`,
+      `--batch=${join(directory, `${dateStamp()}-ballot-%04d.${this.format}`)}`,
       `--batch-print`,
+      `--batch-prompt`,
     ]
 
     debug(
-      'Calling scanimage to scan into %s with prefix=%s in format %s; %s',
+      'Calling scanimage to scan into %s in format %s; %s',
       directory,
-      prefix,
       this.format,
       `scanimage ${args.map((arg) => `'${arg}'`).join(' ')}`
     )
 
-    const onFileScannedPromises: (void | Promise<void>)[] = []
-
-    await new Promise((resolve, reject) => {
-      const scanimage = streamExecFile('scanimage', args)
+    const scannedFiles: string[] = []
+    const results = queue<Promise<IteratorResult<Sheet>>>()
```

Comment:
OMG look at that type structure!! ;)

Reply:
I know, right? Rarely do you get three-level nesting of generics.